### DISCUSSION TITLE
Baremetal instance stuck in starting state and timeout after 30 minutes.

### DISCUSSION THREAD

**Discussion by yummydsky** [apache/cloudstack#10058] [Q&A] [answered]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
Baremetal

##### CLOUDSTACK VERSION
4.19.1.3

##### OS / ENVIRONMENT
Cloudstack 4.19.1.3 on Ubuntu 22.04LTS

##### SUMMARY
I use Ubuntu 22.04 LTS as ISO to deploy a bare metal instance, but it seems that bare metal instance stuck in starting state and timeout like issue https://issues.apache.org/jira/browse/CLOUDSTACK-9716
```
2024-12-02 06:24:44,311 DEBUG [c.c.b.n.BareMetalResourceBase] (DirectAgent-39:ctx-6061c3d0) (logid:47b584a0) still wait for baremetal provision done notification for vm[name:i-2-30-VM], current vm state is Starting
```
![截圖 2024-12-03 上午9 56 12](https://github.com/user-attachments/assets/54902e82-ef21-4edb-a0b9-5f48ef9eaa1c)

##### STEPS TO REPRODUCE
![截圖 2024-12-03 上午9 54 01](https://github.com/user-attachments/assets/5a7ee4ad-ddba-4b60-8e1c-12d3a4f38474)
![截圖 2024-12-03 上午9 54 25](https://github.com/user-attachments/assets/cf1d7ab4-ff04-4bdf-a2f0-1a711a1c94e3)
![截圖 2024-12-03 上午9 54 35](https://github.com/user-attachments/assets/14ea7fe4-6905-4865-8b12-1a4f35367632)

##### EXPECTED RESULTS
The bare metal instance can be deployed os successfully and booted up.

##### ACTUAL RESULTS
The bare metal stuck in starting state because there is no one to notify and change the vm state to `Running`

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by DaanHoogland** [comment]
thanks @yummydsky , baremetal has not been well maintained for a long time. Have you got other bare-metal instances that are running fine?

**Comment by yummydsky** [comment]
> thanks @yummydsky , baremetal has not been well maintained for a long time. Have you got other bare-metal instances that are running fine?

No, there are no bare-metal instances that can be created properly.

**Comment by yummydsky** [comment]
@DaanHoogland Is there a PXE server inside the virtual router? I found a document(http://events17.linuxfoundation.org/sites/events/files/slides/Baremetal-v4.1a.pdf) that said we need an additional VR to act as a PXE server. Do you know how I can add a VR as a PXE server?

**Comment by DaanHoogland** [comment]
@yummydsky I do not know of this , but it seems that adding a pxe server itself to a VR is not a big deal. configuring it properly may vary depending of your infrastructure and requirements.
@rajujith Do you have any experience with this?

**Reply by rajujith** [reply]
@DaanHoogland, I know this feature, but I don't recall trying it. @harikrishna-patnala  may be able to help. However, this feature requires a basic zone based on the document you mentioned? also specific switch model? (http://events17.linuxfoundation.org/sites/events/files/slides/Baremetal-v4.1a.pdf). cc @yummydsky

**Reply by yummydsky** [reply]
@rajujith 
I have no idea about what I need to prepare to give it a try. I assume that there should be a PXE server inside if I choose the network offering that offers the following services.
```
Supported services
BaremetalPxeService: BaremetalPxeProvider
Dhcp: BaremetalDhcpProvider
```

I wonder if I can deploy a bare-metal instance with the `Advanced Zone` because we need to separate the management network and guest network.

If someone can help to provide the details for each step that would be appreciated.

**Comment by yummydsky** [comment]
I found out there is a check if the NetworkType is not Basic, then BareMetal will report `BaremetalDhcpElement can not handle networkoffering: pxe-network`
Does this mean we can only use the basic network for the BareMetal machine?
https://github.com/apache/cloudstack/blob/main/plugins/hypervisors/baremetal/src/main/java/com/cloud/baremetal/networkservice/BaremetalDhcpElement.java#L86

**Comment by DaanHoogland** [comment]
@yummydsky , did you find a resolution to this query?

**Reply by yummydsky** [reply]
No, so far, I have no resolution to this function.

**Reply by DaanHoogland** [reply, marked as answer]
not nice, as I said the baremetal plugin has not been well maintained. There is work going on for a “deploy anything” plugin though (see #9752) and I have seen it being used for baremetal. But this is still experimental and will be in v21 the earliest.
